huaweicloud/services/acceptance/aom/resource_huaweicloud_aom_uniagent_batch_install_test.go

```diff
@@ -0,0 +1,117 @@
+package aom
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccUniAgentBatchInstall_basic(t *testing.T) {
+	rName := acceptance.RandomAccResourceName()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckSecurityGroupName(t)
+			acceptance.TestAccPreCheckAomInstallerAgentId(t)
+			acceptance.TestAccPreCheckAomTestMachineAccount(t)
+			acceptance.TestAccPreCheckAomTestMachinePassword(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		// This resource is a one-time action resource and there is no logic in the delete method.
+		// lintignore:AT001
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testUniAgentBatchInstall_basic(rName),
+			},
+		},
+	})
+}
+
+func testUniAgentBatchInstall_base(rName string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_availability_zones" "test" {}
+
+data "huaweicloud_compute_flavors" "test" {
+  availability_zone = try(data.huaweicloud_availability_zones.test.names[0], "NOT_FOUND")
+  performance_type  = "normal"
+  cpu_core_count    = 2
+  memory_size       = 4
+}
+
+data "huaweicloud_vpc_subnet" "test" {
+  name = "subnet-default"
+}
+
+data "huaweicloud_vpc" "test" {
+  id = data.huaweicloud_vpc_subnet.test.vpc_id
+}
+
+data "huaweicloud_images_image" "test" {
+  name        = "Ubuntu 18.04 server 64bit"
+  most_recent = true
+}
+
+data "huaweicloud_networking_secgroup" "test" {
+  name = "%[1]s"
+}
+
+resource "huaweicloud_compute_instance" "test" {
+  count = 2
+
+  name                = "%[2]s"
+  image_id            = data.huaweicloud_images_image.test.id
+  flavor_id           = try(data.huaweicloud_compute_flavors.test.ids[0], "NOT_FOUND")
+  system_disk_type    = "SAS"
+  system_disk_size    = 40
+  security_group_ids  = [data.huaweicloud_networking_secgroup.test.id] 
+  admin_pass          = "%[3]s"
+  stop_before_destroy = true
+
+  network {
+    uuid = data.huaweicloud_vpc_subnet.test.id
+  }
+
+  tags = {
+    test = "uniagent_install_target"
+  }
+}
+`, acceptance.HW_SECURITY_GROUP_NAME, rName, acceptance.HW_AOM_TEST_MACHINE_PASSWORD)
+}
+
+func testUniAgentBatchInstall_basic(rName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_aom_uniagent_batch_install" "test" {
+  region             = "%[2]s"
+  proxy_region_id    = 0
+  installer_agent_id = "%[3]s"
+  version            = "%[4]s"
+  public_net_flag    = false
+
+  agent_import_param_list {
+    account  = "%[5]s"
+    password = "%[6]s"
+    inner_ip = huaweicloud_compute_instance.test[0].access_ip_v4
+    port     = 22
+    os_type  = "LINUX"
+  }
+
+  agent_import_param_list {
+    account  = "%[5]s"
+    password = "%[6]s"
+    inner_ip = huaweicloud_compute_instance.test[1].access_ip_v4
+    port     = 22
+    os_type  = "LINUX"
+  }
```

Comment:
```suggestion
  dynamic "agent_import_param_list" {
    for_each = huaweicloud_compute_instance.test
  
    content {
      account   = "%[5]s"
      password  = "%[6]s"
      inner_ip  = agent_import_param_list.value.access_ip_v4
      port      = 22
      os_type   = "LINUX"
    }
}
```

recommend using this format

Reply:
done